src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Reference.java

```diff
@@ -1,19 +1,14 @@
 package uk.nhs.digital.nhsconnect.nhais.model.edifact;
 
 import lombok.Getter;
-import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
-import lombok.Setter;
+import lombok.NoArgsConstructor;
 import uk.nhs.digital.nhsconnect.nhais.exceptions.EdifactValidationException;
 
-/**
- *class declaration:
- */
-@Getter @Setter @RequiredArgsConstructor
+@Getter
+@NoArgsConstructor
 public abstract class Reference extends Segment{
-
-    private @NonNull String qualifier;
-    private @NonNull String reference;
+    protected abstract String getReference();
```

Comment:
Reference class has `getReference()` method - looks a little weird

Reply:
Also is `Reference` a `Segment` ? Extending class usually mean "is a" relationship

Reply:
It's and abstract class that has common functionality for all references
The `getReference` is an abstract method to be overriden in by the subclass to provide the value that will be used to. build edifact